core/src/main/java/org/infinispan/metrics/impl/BaseOperatingSystemAdditionalMetrics.java

```diff
@@ -0,0 +1,105 @@
+package org.infinispan.metrics.impl;
+
+import java.lang.management.ManagementFactory;
+import java.lang.management.OperatingSystemMXBean;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.Arrays;
+import java.util.List;
+
+import org.infinispan.util.logging.Log;
+import org.infinispan.util.logging.LogFactory;
+
+import io.micrometer.core.instrument.Gauge;
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.binder.MeterBinder;
+import io.micrometer.core.lang.NonNullApi;
+import io.micrometer.core.lang.NonNullFields;
+
+@NonNullApi
+@NonNullFields
+/**
+ * Provides process metrics.
+ * <p>
+ * Inspired by io.micrometer.core.instrument.binder.system.ProcessorMetrics
+ */
+public class BaseOperatingSystemAdditionalMetrics implements MeterBinder {
+
+   private static final Log log = LogFactory.getLog(BaseOperatingSystemAdditionalMetrics.class);
+
+   private final OperatingSystemMXBean operatingSystemBean;
+   private final Method processCpuLoadMethod;
```

Comment:
This way you guarantee it can only be done once. Say for example if you have multiple instances of this object it doesn't have to do the check multiple times (possibly throwing exceptions and logging error messages etc).

Reply:
Are we expecting more instances? Anyway I'm applying the suggestion. Thanks @wburns!

Reply:
Tests will have thousands :) Also, I am guessing wildfly may need to restart the cache manager in some cases, although it may have a different class loader IDK.

Reply:
right! I didn't think about tests